app/javascript/components/layer.jsx

```diff
@@ -0,0 +1,200 @@
+import React, { useEffect, useRef, useState } from 'react'
+import Paper from 'paper';
+import consumer from '../channels/consumer';
+
+export function Layer(props) {
+  const fontSize = 20;
+
+  const pathRef = useRef(null);
+  const [penState, setPenState] = useState('pen');
+  const [colorState, setColorState] = useState('black');
+  const [pathState, setPathState] = useState([]);
+
+  const sub = (layer, layerId) => {
+    consumer.subscriptions.create({ channel: 'LayerChannel', layer_id: layerId }, {
+      connected() {
+        // Called when the subscription is ready for use on the server
+        console.log(`Connected to layer_channel_${layerId}...`);
+      },
+
+      disconnected() {
+        // Called when the subscription has been terminated by the server
+      },
+
+      received(data) {
+        // Called when there's incoming data on the websocket for this channel
+        layer.addChild(new Paper.PointText({
+          point: [50, 50],
+          content: data,
+          fillColor: props.activeColor,
+          fontSize: fontSize
+        }));
+        console.log(layer);
+      }
+    });
+  }
+
+  useEffect(() => {
+    // Set up action cable subscriber once layer is created
+    if (!!props.layer) {
+      sub(props.layer, props.layerId);
+    }
+  }, [props.layer, props.layerId]);
+
+  useEffect(() => {
+    console.log(props.activeTool)
+    console.log(props.activeColor)
+  }, [props.activeTool, props.activeColor]);
+
+  useEffect(() => {
```

Comment:
I put this here as a placeholder since active tool and active color are changed in the notebook (parent) component so thought we may need to handle something here when that happens but not totally sure. will comment it out until that gets implemented to reduce confusion tho